pkg/sql/opt_catalog.go

```diff
@@ -830,15 +853,26 @@ func (ot *optTable) IsVirtualTable() bool {
 
 // ColumnCount is part of the cat.Table interface.
 func (ot *optTable) ColumnCount() int {
-	return len(ot.desc.DeletableColumns()) + len(ot.systemColumnDescs)
+	return len(ot.desc.DeletableColumns()) + len(ot.systemColumnDescs) + len(ot.virtualColumns)
 }
 
 // Column is part of the cat.Table interface.
 func (ot *optTable) Column(i int) cat.Column {
-	if i >= len(ot.desc.DeletableColumns()) {
-		return ot.systemColumnDescs[i-len(ot.desc.DeletableColumns())]
+	{
+		n := len(ot.desc.DeletableColumns())
+		if i < n {
+			return &ot.desc.DeletableColumns()[i]
+		}
+		i -= n
+	}
+	{
+		n := len(ot.systemColumnDescs)
+		if i < n {
+			return ot.systemColumnDescs[i]
+		}
+		i -= n
 	}
-	return &ot.desc.DeletableColumns()[i]
+	return &ot.virtualColumns[i]
```

Comment:
I would like this to instead match the previous scope block, and then have a panic at the end of this function, similar to ColumnKind below. This makes it easier to identify an error here when a new Column kind/group is added.